src/OrbitDB.js

```diff
@@ -30,17 +31,26 @@ let databaseTypes = {
 class OrbitDB {
   constructor(ipfs, directory, options = {}) {
     this._ipfs = ipfs
-    this.id = options.peerId || (this._ipfs._peerInfo ? this._ipfs._peerInfo.id._idB58String : 'default')
+    this.id = options.peerId
+    this.identity = options.identity
     this._pubsub = options && options.broker
       ? new options.broker(this._ipfs)
       : new Pubsub(this._ipfs, this.id)
-    this.stores = {}
     this.directory = directory || './orbitdb'
-    this.keystore = options.keystore || Keystore.create(path.join(this.directory, this.id, '/keystore'))
-    this.key = this.keystore.getKey(this.id) || this.keystore.createKey(this.id)
+    this.stores = {}
     this._directConnections = {}
   }
 
+  static async createInstance (ipfs, options = {}) {
```

Comment:
Since we're changing the way OrbitDB instances should be created (for the better 👍), we should also update the docs right away. We don't necessarily have to do it in this PR, but should at least leave a TODO to do it immediately after.